src/lib/rest/HttpHeaders.h

```diff
@@ -36,26 +36,37 @@
 *
 * HTTP Headers -
 */
-#define CONTENT_TYPE                 "Content-Type"
-#define FIWARE_SERVICE               "Fiware-Service"
-#define X_AUTH_TOKEN                 "X-Auth-Token"
-#define X_REAL_IP                    "X-Real-IP"
-#define X_FORWARDED_FOR              "X-Forwarded-For"
-#define FIWARE_CORRELATOR            "Fiware-Correlator"
-#define NGSIV2_ATTRSFORMAT           "Ngsiv2-AttrsFormat"
-#define FIWARE_SERVICEPATH           "Fiware-Servicepath"
-#define ACCESS_CONTROL_ALLOW_ORIGIN  "Access-Control-Allow-Origin"
-#define ACCESS_CONTROL_ALLOW_HEADERS "Access-Control-Allow-Headers"
-#define ACCESS_CONTROL_ALLOW_METHODS "Access-Control-Allow-Methods"
-#define ACCESS_CONTROL_MAX_AGE       "Access-Control-Max-Age"
+#define ACCESS_CONTROL_ALLOW_ORIGIN   "Access-Control-Allow-Origin"
+#define ACCESS_CONTROL_ALLOW_HEADERS  "Access-Control-Allow-Headers"
+#define ACCESS_CONTROL_ALLOW_METHODS  "Access-Control-Allow-Methods"
+#define ACCESS_CONTROL_MAX_AGE        "Access-Control-Max-Age"
+#define ACCESS_CONTROL_EXPOSE_HEADERS "Access-Control-Expose-Headers"
+#define CONTENT_TYPE                  "Content-Type"
+#define FIWARE_CORRELATOR             "Fiware-Correlator"
+#define FIWARE_SERVICE                "Fiware-Service"
+#define FIWARE_SERVICEPATH            "Fiware-Servicepath"
+#define FIWARE_TOTAL_COUNT            "Fiware-Total-Count"
+#define NGSIV2_ATTRSFORMAT            "Ngsiv2-AttrsFormat"
+#define RESOURCE_LOCATION             "Location"
+#define X_AUTH_TOKEN                  "X-Auth-Token"
+#define X_REAL_IP                     "X-Real-IP"
+#define X_FORWARDED_FOR               "X-Forwarded-For"
 
 
 
 /* ****************************************************************************
 *
 * CORS Allowed Headers -
 */
-#define CORS_ALLOWED_HEADERS CONTENT_TYPE", "FIWARE_SERVICE", "FIWARE_SERVICEPATH", "NGSIV2_ATTRSFORMAT", "FIWARE_CORRELATOR", "X_FORWARDED_FOR", " X_REAL_IP", " X_AUTH_TOKEN
+#define CORS_ALLOWED_HEADERS CONTENT_TYPE", "FIWARE_SERVICE", "FIWARE_SERVICEPATH", "NGSIV2_ATTRSFORMAT", "FIWARE_CORRELATOR", "X_FORWARDED_FOR", "X_REAL_IP", "X_AUTH_TOKEN
+
+
+
+/* ****************************************************************************
+*
+* CORS Exposed Headers -
+*/
+#define CORS_EXPOSED_HEADERS FIWARE_CORRELATOR", "FIWARE_TOTAL_COUNT", "RESOURCE_LOCATION
```

Comment:
Should the three exposed headers be documented as part of Orion documentation? I mean, is this list something implementation-specific (so it should) or derived directly from CORS specification without ambiguity?

Reply:
Well, these are the headers we put in Context Broker responses that the client could read. In the context of CORS, we need to have these headers listed in Access-Control-Expose-Headers in our responses so that the client can access and read them. 

Since we are not limiting or extending this set for CORS, I would say documenting might not be crucial but won't hurt either : ) 

By the way, if I have missed any headers that we put in the responses and expect it to be client accessible please let me know.

Reply:
> Since we are not limiting or extending this set for CORS, I would say documenting might not be crucial but won't hurt either : )

I agree.

Could you write a little cors.md piece of documentation about this (and any other CORS specific aspects to mention) in doc/manuals/user to be included in this PR, please? Don't worty too much about how the material fits in the overall manual, I'll deal with it once the PR gets merged.

Reply:
Fixed in 3e6e50a